### PR TITLE
 Add database migrations

### PR DESCRIPTION
**Pull request by nozim**
- added database migrations and golang migrate library 
- added go migrate library installation script 
- added separate docker compose for local arc and postgres database

### CODE REVIEW

File: database/migrations/000001_create_blocks.up.sql [hunk 1]
```diff
@@ -0,0 +1,13 @@
+CREATE TABLE blocks (
+                        inserted_at timestamp with time zone DEFAULT CURRENT_TIMESTAMP NOT NULL,
```
Comment (boecklim) [inline database/migrations/000001_create_blocks.up.sql:2]: IMO all sql keywords should be capitalised

File: database/schema.sql 
Comment (boecklim) [inline database/schema.sql :1]: What is this file used for? Is it the schema for sqlite? Couldn't find any reference to it.

Comment (nozim) [reply database/schema.sql :1]: this is in case if you need to run the current schema all at once. But can be removed as well :)

**Comment by nozim** [comment]
@boecklim for sqlite migrations I was thinking if we could reuse the current ones. The should be universal.

File: scripts/install-migrate.sh
[the remark this answers is not in the document]
Comment (nozim) [reply scripts/install-migrate.sh:1]: renamed in latest commit.

**Review by boecklim — APPROVED** [review]

**Pull request merged**